libs/@local/harpc/service/examples/account.rs

```diff
@@ -0,0 +1,195 @@
+#![feature(never_type)]
+#![expect(
+    dead_code,
+    clippy::unwrap_used,
+    clippy::empty_enum,
+    clippy::todo,
+    reason = "non-working example code"
+)]
+
+use error_stack::Report;
+use frunk::HList;
+use futures::{StreamExt, pin_mut, stream};
+use graph_types::account::AccountId;
+use harpc_net::codec::{Decoder, Encoder};
+use harpc_service::{
+    Service,
+    delegate::ServiceDelegate,
+    metadata::Metadata,
+    procedure::{Procedure, ProcedureIdentifier},
+    role::{Client, ClientSession, Role, Server},
+};
+use harpc_tower::{
+    body::{Body, BodyExt},
+    either::Either,
+    request::Request,
+    response::{Parts, Response},
+};
+use harpc_types::{procedure::ProcedureId, service::ServiceId, version::Version};
+use harpc_wire_protocol::{request::procedure::ProcedureDescriptor, response::kind::ResponseKind};
+
+enum AccountProcedureId {
+    CreateAccount,
+}
+
+impl ProcedureIdentifier for AccountProcedureId {
+    fn from_id(id: ProcedureId) -> Option<Self> {
+        match id.value() {
+            0 => Some(Self::CreateAccount),
+            _ => None,
+        }
+    }
+
+    fn into_id(self) -> ProcedureId {
+        match self {
+            Self::CreateAccount => ProcedureId::new(0),
+        }
+    }
+}
+
+struct Account;
+
+impl Service for Account {
+    type ProcedureId = AccountProcedureId;
+    type Procedures = HList![CreateAccount];
+
+    const ID: ServiceId = ServiceId::new(0);
+    const VERSION: Version = Version {
+        major: 0x00,
+        minor: 0x00,
+    };
+
+    fn metadata() -> Metadata {
+        Metadata {
+            since: Version {
+                major: 0x00,
+                minor: 0x00,
+            },
+            deprecation: None,
+            removal: None,
+        }
+    }
+}
+
+struct CreateAccount {
+    id: Option<AccountId>,
+}
+
+impl Procedure for CreateAccount {
+    type Service = Account;
+
+    const ID: <Self::Service as Service>::ProcedureId = AccountProcedureId::CreateAccount;
+
+    fn metadata() -> Metadata {
+        Metadata {
+            since: Version {
+                major: 0x00,
+                minor: 0x00,
+            },
+            deprecation: None,
+            removal: None,
+        }
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, thiserror::Error)]
+enum AccountError {}
+
+trait AccountService<R>
+where
+    R: Role,
+{
+    fn create_account(
+        &self,
+        session: &R::Session,
+        payload: CreateAccount,
+    ) -> impl Future<Output = Result<AccountId, Report<AccountError>>> + Send;
+}
+
+struct AccountServiceImpl;
+
+impl<S> AccountService<Server<S>> for AccountServiceImpl
+where
+    S: Send + Sync,
+{
+    async fn create_account(
+        &self,
+        _: &S,
+        _: CreateAccount,
```

Comment:
Can we add names to variables here?

Reply:
see: [d3cdbb9](https://github.com/hashintel/hash/pull/5283/commits/d3cdbb9bc463ff11427b80ff6103fb04c00e65e3)